### PR TITLE
feat(test-utils): testing for dev

### PR DESCRIPTION
**Pull request by antfu**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
@antfu Do you mind checking failing tests?

File: packages/test-utils/src/dev.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  ctx.url = 'http://localhost:' + port
+
+  let serverReadyResolve = () => {}
+  const serverReadyPromise = new Promise<void>((resolve) => {
+    serverReadyResolve = resolve
+  })
+
+  const stdout = new Writable({
+    write (data, encoding, callback) {
+      // console.log({ out: data.toString() })
```
Comment (antfu) [inline packages/test-utils/src/dev.ts:18]: If you comment this out, you will see
image

I can confirm that commands/dev.ts get invoked but not sure why it goes to build mode. Any clue @pi0?

Comment (pi0) [reply packages/test-utils/src/dev.ts:18]: Sorry i missed this message 🙈  Will locally check

Comment (pi0) [reply packages/test-utils/src/dev.ts:18]: Seems related to https://github.com/nuxt/framework/issues/3262

Simpler reproduction: 

```js
import nuxi from 'nuxi'
await nuxi.runCommand('dev', ['./examples/hello-world'])
```

image

Comment (pi0) [reply packages/test-utils/src/dev.ts:18]: https://github.com/nuxt/framework/pull/3751

**Comment by pi0** [comment]
Closing in favor of #3753 to directly use Nuxt instance. Sorry for troubles!

**Pull request closed without merging**